### PR TITLE
[passport-google-oauth20] Extend Strategy constructor overloads

### PR DESCRIPTION
**Pull request by d4ncer**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: [verify with params](https://github.com/jaredhanson/passport-oauth2/blob/master/lib/strategy.js#L193)
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

--

This change adds constructor overloads for the cases where the consumer is interested in accessing the params passed back to be verified by the Google OAuth 2.0 strategy. Relevant source code:

```js
if (self._passReqToCallback) {
  var arity = self._verify.length;
  if (arity == 6) {
    self._verify(req, accessToken, refreshToken, params, profile, verified); // THIS
  } else { // arity == 5
    self._verify(req, accessToken, refreshToken, profile, verified);
  }
} else {
  var arity = self._verify.length;
  if (arity == 5) {
    self._verify(accessToken, refreshToken, params, profile, verified); // AND THIS
  } else { // arity == 4
    self._verify(accessToken, refreshToken, profile, verified);
  }
}
```

### CODE REVIEW

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## passport-google-oauth20 ([<kbd>unpkg</kbd>](https://unpkg.com/browse/passport-google-oauth20@latest/))
was missing the following properties:
1. The declaration doesn't match the JavaScript module 'passport-google-oauth20'. Reason:
The JavaScript module can be called or constructed, but the declaration module cannot.

The most common way to resolve this error is to use 'export =' syntax.
To learn more about 'export =' syntax, see https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 614289f0efc7886a4699cd19de21e5649ded6646
</p>

File: types/passport-google-oauth20/index.d.ts [hunk 1]
```diff
@@ -7,10 +7,9 @@
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 2.8
 
-import * as passport from "passport";
-import * as express from "express";
-import * as oauth2 from "passport-oauth2";
-import { OutgoingHttpHeaders } from "http";
+import passport = require("passport");
```
Comment (ezintz) [inline types/passport-google-oauth20/index.d.ts:10]: Why did you change the imports? I kinda see no benefit?

Comment (d4ncer) [reply types/passport-google-oauth20/index.d.ts:10]: I saw this import pattern used across other type definition files, so figured I'd make this consistent. Happy to revert this.

Comment (d4ncer) [reply types/passport-google-oauth20/index.d.ts]: @ezintz reverted.
